### PR TITLE
*: enable lint test.

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref #11606

### What is changed and how it works?
enable lint checks that was temp disable due to upgrade go to 1.23.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 [x] No code

### Release note 

```release-note
N/A
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]
we can see only dm uses `"github.com/pingcap/check"`. we should replace it with testify gradually. https://github.com/pingcap/tidb/issues/26022

**Review by 3AceShowHand — APPROVED** [review]

**Comment by 3AceShowHand** [comment]
the cdc part LGTM

**Review by wlwilliamx — APPROVED** [review]
cdc part LGTM

**Review by purelind — APPROVED** [review]
/lgtm

File: pkg/applier/redo.go [hunk 1]
```diff
@@ -615,6 +615,7 @@ func processEvent(
 			return nil, event, nil
 		}
 	}
+	// nolint
```
[the remark this answers is not in the document]
Comment (Benjamin2037) [reply pkg/applier/redo.go:618]: This nolint is used for revive check，if last else with return or break，suggest to remove  else part.

File: pkg/sink/codec/builder/codec_test.go [hunk 2]
```diff
@@ -99,6 +99,7 @@ func codecEncodeKeyPB(event *model.RowChangedEvent) []byte {
 		RowId:     event.RowID,
 		Partition: 0,
 	}
+	// nolint
```
[the remark this answers is not in the document]
Comment (Benjamin2037) [reply pkg/sink/codec/builder/codec_test.go:102]: This is also revive check, suggest to remove else part of if statement.

**Review by yudongusa — APPROVED** [review]
LGTM

**Pull request merged**